docs/service_architecture.md

```diff
@@ -166,6 +169,87 @@ keys in the HSM, but not to exfiltrate the private material.
   signing arbitrary metadata, with the important detail that in this particular
   case key rotations will be necessary to recover from the attack.
 
+### Notary client keys and credentials compromise
+
+In the event of a trust collection owner/administrator's keys being compromised,
+an attacker would be able to sign valid changes to the contents of that
+collection, and to any other collections that use the same key.
+
+They can then set up a mirrior to distribute this metadata, but they would not
+be able to distribute it via a Notary service unless they also have write-capable
+credentials for that service (e.g. the username/password of a user who could
+push updates into that service).
+
+A familiarity with [TUF (the update framework)](theupdateframework.github.io)
+would be helpful in understanding the different types of keys and roles
+mentioned below.
+
+Note that the descriptions below assume that the snapshot key is managed by the
+Notary service - otherwise, in addition to the keys below, the snapshot key would
+also have to be compromised in order to perform any of the attacks.
+
+#### Delegation key compromise
+
+A delegation key has the most limited capabilities of any client-managed key.
+It is used to sign targets into [specific delegation roles, which may have path
+restrictions](advanced-usage.md#work-with-delegation-roles), and can further
```

Comment:
I think this is a broken link, should be `advanced_usage.md#work-with-delegation-roles` ?